artificial-scrum-master/apps/artificial.scrum.master/Artificial.Scrum.Master.User/UserModule.cs

```diff
@@ -0,0 +1,115 @@
+using Artificial.Scrum.Master.User.Features.Authorization;
+using Artificial.Scrum.Master.User.Features.ExternalSignIn;
+using Artificial.Scrum.Master.User.Infrastructure.JWT;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.IdentityModel.Tokens;
+using System.Text;
+
+namespace Artificial.Scrum.Master.User
+{
+    public static class UserModule
+    {
+        public static void AddUserModule(this IServiceCollection services, IConfiguration configuration)
+        {
+            var encryptionKey = configuration["Jwt:Key"] ?? throw new ArgumentNullException("Jwt:Key");
+
+            services.AddTransient<IExternalSignInService, ExternalSignInService>();
+            services.AddTransient<IJwtAuthorizationService, JwtAuthorizationService>();
+            services.AddTransient<IAuthorizationService, AuthorizationService>();
+            services.AddAuthentication(x =>
+            {
+                x.DefaultAuthenticateScheme = JwtBearerDefaults.AuthenticationScheme;
+                x.DefaultChallengeScheme = JwtBearerDefaults.AuthenticationScheme;
+            })
+                .AddJwtBearer(x =>
+                {
+                    x.RequireHttpsMetadata = false;
+                    x.SaveToken = true;
+                    x.TokenValidationParameters = new TokenValidationParameters
+                    {
+                        ValidateIssuerSigningKey = true,
+                        IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(encryptionKey)),
+                        ValidateIssuer = false,
+                        ValidateAudience = false,
+                    };
+
+                    x.Events = new JwtBearerEvents
+                    {
+                        OnMessageReceived = context =>
+                        {
+                            context.Token = context.Request.Cookies["AuthToken"];
```

Comment:
może to ciastko jako const w klasie?

Reply:
nope, przy ciastkach i wyciaganiu z requestu lepiej nie wyciagac rzecy do constow a tym bardziej do jakichs w klasie, bo nie wiadomo jak sie to do konca zachowa